packages/@aws-cdk/cdk/lib/cloudformation/output.ts

```diff
@@ -16,21 +16,22 @@ export interface OutputProps {
   value?: any;
 
   /**
-   * The name used to export the value of this output across stacks. To import
-   * the value from another stack, use `FnImportValue(export)`. You can create
-   * an import value token by calling `output.makeImportValue()`.
+   * The name used to export the value of this output across stacks.
    *
-   * @default The default behavior is to automatically allocate an export name
-   * for outputs based on the stack name and the output's logical ID. To
-   * create an output without an export, set `disableExport: true`.
+   * To import the value from another stack, use `FnImportValue(export)`. You
+   * can create an import value token by calling `output.makeImportValue()`.
+   *
+   * @default Automatically allocate a name when `makeImportValue()`  is
+   * called.
    */
   export?: string;
 
   /**
    * Disables the automatic allocation of an export name for this output.
    *
-   * @default false, which means that an export name is either explicitly
-   * specified or allocated based on the output's logical ID and stack name.
+   * This disables use of `makeImportValue()` if `export` is  not given.
```

Comment:
I would say `prohibits` rather than `disables` here... What is disabled is *auto-allocation of an export name*, and the corollary is you cannot use `makeImportValue()` unless an `export` name was provided at creation time.